magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -0,0 +1,571 @@
+"""
+This script splits the proton MC data sample into "train"
+and "test", deletes possible failed runs (only those files
+that end up with a size < 1 kB), and generates the bash
+scripts to merge the data files calling the script "merge_hdf_files.py"
+in the following order:
+
+MAGIC:
+1) Merge the subruns into runs for M1 and M2 individually.
+2) Merge the runs of M1 and M2 into M1-M2 runs.
+3) Merge all the M1-M2 runs for a given night.
+Workingdir/DL1/Observations/Merged
+
+MC:
+1) Merges all MC runs in a node and save them at
+Workingdir/DL1/MC/PARTICLE/Merged
+
+Usage:
+$ python merging_runs_and_splitting_training_samples.py (-c config.yaml)
+
+If you want to merge only the MAGIC or only the MC data,
+you can do as follows:
+
+Only MAGIC:
+$ python merging_runs_and_splitting_training_samples.py --analysis-type onlyMAGIC (-c config.yaml)
+
+Only MC:
+$ python merging_runs_and_splitting_training_samples.py --analysis-type onlyMC (-c config.yaml)
+"""
+
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+from tqdm import tqdm
+
+from magicctapipe import __version__
+
+__all__ = ["cleaning", "split_train_test", "merge", "mergeMC"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def cleaning(list_of_nodes, cwd):
+
+    """
+    This function looks for failed runs in each node and remove them.
+
+    Parameters
+    ----------
+    list_of_nodes : array of str
+        List of nodes where the function will look for failed runs.
+    cwd : Path
+        Current working directory
+    """
+
+    for i in tqdm(range(len(list_of_nodes)), desc="Cleaning failed runs"):
+        os.chdir(list_of_nodes[i])
+        os.system('find . -type f -name "*.h5" -size -1k -delete')
+
+    os.chdir(cwd)
+    print("Cleaning done.")
+
+
+def split_train_test(target_dir, train_fraction, source_name):
+
+    """
+    This function splits the MC proton sample in 2, i.e. the "test" and the "train" subsamples.
+    It generates 2 subdirectories in the directory .../DL1/MC/protons named "test" and "train" and creates sub-sub-directories with the names of all nodes.
+    For each node sub-sub-directory we move 80% of the .h5 files (if it is in the "test" subdirectory) or 20% of the .h5 files (if it is in the "train" subdirectory).
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    train_fraction : float
+        Fraction of proton MC files to be used in the training RF dataset
+    source_name : str
```

Comment:
why do you add here source_name (that is later on added to the path?
you might have separate MCs for different data sets, but this depends on the date (analysis period) and on NSB level), and not on actual source. Namely you can have the same source observed in dark time and mild moon with different settings, and they would end up in the same directory. On the other hand a bunch of extragal sources from the same period observed in dark time can share the same MCs.

Reply:
I will try to move the MCs out of the source directories: we can use these scripts to run a single MC production ( 1 NSB, 1 period, 1 declination...) so we can make it source independent

Reply:
And, as already discussed, this MC processing will disappear before releasing the new version of the pipeline